src/controllers/main/main.ts

```diff
@@ -627,19 +461,26 @@ export class MainController extends EventEmitter {
     )
   }
 
-  async updateSelectedAccount(selectedAccount: string | null = null, forceUpdate: boolean = false) {
-    if (!selectedAccount) return
+  async updateSelectedAccountPortfolio(forceUpdate: boolean = false) {
+    if (!this.accounts.selectedAccount) return
 
     // pass the accountOps if any so we could reflect the pending state
     const accountOps =
       this.actions.currentAction?.type === 'accountOp' && this.signAccountOp
         ? {
             [this.actions.currentAction.accountOp.networkId]: [this.actions.currentAction.accountOp]
           }
-        : getAccountOpsByNetwork(selectedAccount, this.actions.visibleActionsQueue)
-    this.portfolio.updateSelectedAccount(this.accounts, selectedAccount, undefined, accountOps, {
-      forceUpdate
-    })
+        : getAccountOpsByNetwork(this.accounts.selectedAccount, this.actions.visibleActionsQueue)
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    this.portfolio.updateSelectedAccount(
+      this.accounts.accounts,
```

Comment:
We can pass the `accounts` controller to `portfolio`'s constructor so we don't have to pass `this.accounts.accounts` on every update(the same we did for networks)